extensions/blocks/contact-form/components/jetpack-newsletter-integration-settings-creativemail.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * External dependencies
+ */
+import { Spinner, Notice } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { get } from 'lodash';
+import { useCallback, useState, useEffect } from '@wordpress/element';
+import { getPlugins } from './../../../shared/plugin-management';
+
+import CreativeMailPluginErrorState from './jetpack-newsletter-integration-settings-error-state';
+import CreativeMailPluginState, {
+	pluginStateEnum,
+} from './jetpack-newsletter-integration-settings-plugin-state';
+
+const pluginPathWithoutPhp = 'creative-mail-by-constant-contact/creative-mail-plugin';
+const pluginPath = `${ pluginPathWithoutPhp }.php`;
+
+const useOnCreativeMailPluginPromise = ( setPluginError, setIsInstalling, setPluginState ) => {
+	const onCreativeMailPluginClick = useCallback(
+		( func, arg ) => {
+			setPluginError( undefined );
+			setIsInstalling( true );
+			func( arg )
+				.then( () => {
+					setPluginState( pluginStateEnum.ACTIVE );
+				} )
+				.catch( err => {
+					setPluginError( err );
+				} )
+				.finally( () => setIsInstalling( false ) );
+		},
+		[ setIsInstalling, setPluginError, setPluginState ]
+	);
+	return onCreativeMailPluginClick;
+};
+
+const CreativeMailPluginFetched = ( { pluginState, setPluginState } ) => {
+	const [ pluginError, setPluginError ] = useState();
+	const [ isInstalling, setIsInstalling ] = useState( false );
+	const onCreativeMailPluginClick = useOnCreativeMailPluginPromise(
+		setPluginError,
+		setIsInstalling,
+		setPluginState
+	);
+	if ( pluginError ) {
+		return <CreativeMailPluginErrorState error={ pluginError } />;
+	}
+	return (
+		<CreativeMailPluginState
+			pluginState={ pluginState }
+			onCreativeMailPluginClick={ onCreativeMailPluginClick }
+			isInstalling={ isInstalling }
+		/>
+	);
+};
+
+const CreativeMailPluginsLoaded = ( {
+	isFetchingPlugins,
+	hasError,
+	pluginState,
+	setPluginState,
+} ) => {
+	if ( isFetchingPlugins ) {
+		return <Spinner />;
+	}
+	if ( hasError ) {
+		return (
+			<Notice isDismissible={ false } status="error">
+				{ __( "Couldn't access the plugins. Please try again later.", 'jetpack' ) }
```

Comment:
I feel like this message isn't very clear and since the user does not have sufficient permissions it is not actionable for them. I'm not sure if we should display anything here at all in this case. Thoughts @jeherve ?

Reply:
Right, if someone does not have access to plugins, I do not think we should display anything.

Reply:
I got this message from the CRM integration , do you prefer to not show anything?

Reply:
I remove the error message for now.